scipy/stats/stats.py

```diff
@@ -3818,6 +3828,9 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate'):
     (0.052760927029710199, 0.60213045837062351)
 
     """
+    if axis is not None and axis > 1:
+        raise ValueError("spearmanr only handles 1-D or 2-D arrays")
```

Comment:
Can you add a statement like "please use only values 0, 1 or None for axis"? Otherwise the user might think that x or y are not the right input

Reply:
Ok, I'll see about adding something like: 
```
raise ValueError("spearmanr only handles 1-D or 2-D arrays, supplied axis argument {}, please use only values 0, 1 or None for axis".format(axis))
```

This is helpful because you might be looking at logs that were stored from calling on a different host and perhaps a long time prior and no longer have the session state, e.g. you won't know what value was passed to `axis`. This log tells what argument value caused the error as well as how to fix the error. It is a bit long, but this was as short as I could conjure with both violating value and note of acceptable values. Open to changes for brevity if you have a message w/briefer string literal that conveys failing value and acceptable values.